aep/general/0122/aep.md

```diff
@@ -0,0 +1,318 @@
+  - If Unicode characters can not be avoided, resource paths **must** be stored
+    in Normalization Form C (see [AIP-210][]).
+- Eacg resource **must** expose a `path` field that contains its resource path.
+  - Resources **may** provide the resource ID, i.e. the last segment of the
+    path, as a separate field named `id`.
+  - Resources **may** expose a separate, system-generated unique ID field
+    (`uid`).
+  - Resources **must not** expose tuples, self-links, or other forms of
+    resource identification.
+  - All ID fields **must** be strings.
+
+**Note:** Resource paths as described here are used within the scope of a
+single API (or else in situations where the owning API is clear from the
+context), and are only required to be unique within that scope. For this
+reason, they are sometimes called _relative resource paths_ to distinguish them
+from _full resource paths_ (discussed below).
+
+[aip-210]: ./0210.md
+[uri path schema]: https://datatracker.ietf.org/doc/html/rfc3986#appendix-A
+
+### Collection identifiers
+
+The collection identifier segments in a resource path **must** be the plural
+form of the noun used for the resource. (For example, a collection of
+`Publisher` resources is called `publishers` in the resource path.)
+
+- Collection identifiers **must** be concise American English terms.
+- Collection identifiers **must** be in `kebab-case`.
+- Collection identifiers **must** begin with a lower-cased letter and contain
+  only lower-case ASCII letters, numbers. and hyphens (`/[a-z][a-z0-9-]*/`).
+- Collection identifiers **must** be plural.
+  - In situations where there is no plural word ("info"), or where the singular
+    and plural terms are the same ("moose"), the non-pluralized (singular) form
+    is correct. Collection segments **must not** "coin" words by adding "s" in
+    such cases (e.g. avoid "infos").
+
+#### Nested collections
+
+If a resource path contains multiple levels of a hierarchy, and a parent
+collection's path is used as a prefix for the child resource's path, the child
+collection's path **may** omit the prefix. For example, given a collection of
+`UserEvent` resources that would normally be nested underneath `users`:
+
+```
+users/vhugo1802/user-events/birthday-dinner-226/user-event-guests/123
+```
+
+An API **may** use the less-redundant form:
+
+```
+users/vhugo1802/events/birthday-dinner-226/guests/123
+```
+
+In this situation, the _message_ is still called `UserEvent` or
+`UserEventGuest`; only the collection name is shortened.
+
+**Note:** APIs wishing to do this **must** follow this format consistently
+throughout the API, or else not at all.
+
+### Resource ID segments
+
+A resource ID segment identifies the resource within its parent collection. In
+the resource path `publishers/123/books/les-miserables`, `123` is the resource
+ID for the publisher, and `les-miserables` is the resource ID for the book.
+
+- Resource IDs **may** be either always set by users (required on resource
```

Comment:
Good question.  I guess it sets context for the sub-bullets?  And makes it unambiguous that all variants are considered acceptable by the AIPs (readers might make assumptions otherwise).

But that's me playing devil's advocate for keeping it, not actually strongly arguing for doing so.  Do you have a suggested rephrasing/refactor for this bullet and its sub-bullets?

Reply:
I kind of feel like it should be removed entirely, but on the other hand I do agree that clarifying the options is good. I'll just resolve this for now.

My main concern here is needlessly verbose guidance - it make the spec harder to read.